### ISSUE TITLE
Ability to generate parameterized queries/prepared statements

### ISSUE THREAD

**Issue by mwozniczak** [react-querybuilder/react-querybuilder#138]
Hi, I was looking into this library during a refactor of a legacy, jquery-based frontend. This one looks pretty similar to one of the components there, but I can't seem to find the functionality that would output a parameterized query, that would be safer to execute, similar to what the original does [here](https://querybuilder.js.org/plugins.html) (search for '`question_mark`' to see what I mean, specifically)

I need to be able to get the query in a format similar to something like

```javascript

{
  sql: 'author LIKE ? AND publish_date BETWEEN ? AND ?',
  params: ['%Tolkien%', '1954/01/01', '1956/01/01']
}
```

Can this library do something like this?

**Comment by jakeboone02** [comment]
This library does not currently output parameterized SQL, but it would be a fairly simple addition to the `formatQuery` function.  I may be able to get to it today or tomorrow, unless you want to put in a PR to address it yourself.

**Comment by jakeboone02** [comment]
@mwozniczak I opened #139 to add the requested functionality.  Have a look and see if it meets your needs.  If so, I'll merge it and publish a new version.

**Comment by mwozniczak** [comment]
@jakeboone02 looking at the tests, it certainly seems like it! that was pretty fast, too!

**Comment by mwozniczak** [comment]
@jakeboone02 okay, quick update, I've got a test controller where I'm doing something like this:

```javascript
    console.log('Query is now', query)
    console.log('It will be', formatQuery(query, 'parameterized'), typeof formatQuery(query, 'parameterized'))
    console.log('Which maps to', formatQuery(query, 'sql'))
```

Weirdly enough, the results I'm getting in console.log are:

```
Query is now {id: "g-3IJAMjOJBJaEVje166BBU", rules: Array(1), combinator: "or", not: false}
App.js:59 It will be  string
App.js:60 Which maps to (some_field like "%aa%")
```

so it seems like for some reason this call: `formatQuery(query, 'parameterized')` is returning an empty string.

I'm not entirely sure why that is, as from the test suite you have it's parsed correctly, AND is an object.

Edit: FWIW, my import looks like this:

```javascript
import QueryBuilder, {formatQuery} from 'react-querybuilder'
```

and the entry in package.json is currently
```json
"react-querybuilder": "git://github.com/sapientglobalmarkets/react-querybuilder.git#parameterized-queries",
```

**Comment by jakeboone02** [comment]
Oops!  I forgot to update the `dist` and `demo` folders after adding the feature.  Try again now.

**Comment by mwozniczak** [comment]
@jakeboone02 yup, that seems to have fixed it, all good now!

**Comment by jakeboone02** [comment]
@mwozniczak this feature has been released as part of [v3.2.0](https://github.com/sapientglobalmarkets/react-querybuilder/releases/tag/v3.2.0).

**Comment by mwozniczak** [comment]
@jakeboone02 yup, upgraded to it today, and it works like a charm!
